R/textplot_wordcloud.R

```diff
@@ -1,70 +1,468 @@
+                r <- r + r_step * theta_step / (2 * pi)
+                x1 <- 0.5 + r * cos(theta)
+                y1 <- 0.5 + r * sin(theta)
+            }
+        }
+    }
+    #abline(v=c(0, 0.25, 0.75, 1), h=c(0, 0.25, 0.75, 1))
+    graphics::par(op)
+}
+
+#' Internal function for textplot_wordcloud
+#'
+#' This function impliments wordcloud that compares documents. Code is adopted from
+#' \code{\link[wordcloud]{comparison.cloud}}.
+#' @inheritParams textplot_wordcloud
+#' @param scale deprecated argument
+#' @param min.freq deprecated argument
+#' @param max.words deprecated argument
+#' @param random.order deprecated argument
+#' @param random.color  deprecated argument
+#' @param rot.per deprecated argument
+#' @param ordered.colors deprecated argument
+#' @param use.r.layout deprecated argument
+#' @param title.size deprecated argument
+#' @keywords internal
+#' @author Ian Fellows
+wordcloud_comparison <- function(x, min_size, max_size, max_words,
+                                 color, font, adjust, rotation,
+                                 random_order, random_color, ordered_color,
+                                 labelcolor, labelsize, labeloffset, fixed_aspect,
+                                 # deprecated arguments
+                                 colors, scale, min.freq, max.words, 
+                                 random.order, rot.per, use.r.layout, title.size,
+                                 ...) {
+    
+    if (!missing(min.freq)) {
+        warning('min.freq is deprecated; use dfm_trim() before textplot_wordcloud()', call. = FALSE)
+        x <- dfm_trim(x, min_count = min.freq)
+    }
+    arg_dep <- character()
+    if (!missing(colors)) {
+        color <- colors
+        arg_dep <- c(arg_dep, 'color' = 'colors')
+    }
+    if (!missing(scale)) {
+        max_size <- scale[1]
+        min_size <- scale[2]
+        arg_dep <- c(arg_dep, 'min_size and max_size' = 'scale')
+    }
+    if (!missing(max.words)) {
+        max_words <- max.words
+        arg_dep <- c(arg_dep, 'max_words' = 'max.words')
+    }
+    if (!missing(random.order)) {
+        random_order <- random.order
+        arg_dep <- c(arg_dep, 'random_order' = 'random.order')
+    }
+    if (!missing(rot.per)) {
+        rotation <- rot.per
+        arg_dep <- c(arg_dep, 'rotation' = 'rot.per')
+    }
+    if (!missing(use.r.layout)) {
+        warning('use.r.layout is no longer used', call. = FALSE)
+    }
+    if (!missing(title.size)) {
+        labelsize <- title.size
+        arg_dep <- c(arg_dep, 'labelsize' = 'title.size')
+    }
+    if (length(arg_dep)) {
+        warning(paste(arg_dep), " is deprecated; use ", paste(names(arg_dep)), " instead", call. = FALSE)
+    }
+    
+    font <- check_font(font)
+    x <- dfm_weight(x, 'prop')
+    x <- t(x) - Matrix::colMeans(x)
+    x <- as.matrix(x)
+
+    ndoc <- ncol(x)
+    theta_bins <- seq(0, 2 * pi, length = ndoc + 1)
+
+    if (is.null(color) < ndoc)
```

Comment:
This is an error since the condition will always be true. Should this be `length(color)`? 

In calling the function, I am not able to change the colour when `comparison = TRUE`, and this statement is why.

Also, let's put the default color as a scheme in the function signature, not buried in the function.

Reply:
Fixed the bug, but not sure what to do with default colors. If it is not 'Paired', 'Accent' is good because its first two colors are very different (gray and green). This nonetheless goes away from setting a consistent color scheme.